### PR TITLE
[Amara] Add new extractor

### PR DESCRIPTION
**Pull request by joostverdoorn**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR adds support for videos on the amara.org website. Amara allows users to add subtitles to Youtube and Vimeo videos, even though they might not be the uploader of that video on Youtube or Vimeo. Amara is often used in educational settings where teachers need to make an existing video accessible for hearing-impaired students.

### CODE REVIEW

**Comment by lhoBas** [comment]
@dstftw I'd love to start using this, any remaining steps to take for a merge? Thanks 😄

**Comment by joostverdoorn** [comment]
@remitamine @dstftw I understand you're probably swamped, but I'd love to get this shipped, is there anything holding this PR back? :)

File: youtube_dl/extractor/amara.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+            }
+        },
+        {
+            'url': 'https://amara.org/en/videos/s8KL7I3jLmh6/info/the-danger-of-a-single-story/',
+            'md5': 'd3970f08512738ee60c5807311ff5d3f',
+            'info_dict': {
+                'id': 's8KL7I3jLmh6',
+                'ext': 'mp4',
+                'title': 'The danger of a single story',
+                'description': 'Our lives, our cultures, are composed of many overlapping stories. Novelist Chimamanda Adichie tells the story of how she found her authentic cultural voice -- and warns that if we hear only a single story about another person or country, we risk a critical misunderstanding.',
```
Comment (dstftw) [inline youtube_dl/extractor/amara.py]: `'md5:'`.

File: youtube_dl/extractor/amara.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+                    ]
+                },
+                'upload_date': '20131206'
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        meta = self._download_json('https://amara.org/api/videos/%s/?extra=player_urls&format=json' % video_id, video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/amara.py]: Move query to `query`.

**Comment by joostverdoorn** [comment]
@dstftw thanks for the review! Your comments have been processed.

**Comment by joostverdoorn** [comment]
@dstftw would it be possible to get this merged? :)

**Comment by joostverdoorn** [comment]
@dstftw would it be possible to get this merged now? :)

File: youtube_dl/extractor/amara.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        ], filter(lambda language: language['published'], meta.get('languages', []))))
+
+        return {
+            '_type': 'url_transparent',
+            'url': video_url,
+            'id': video_id,
+            'subtitles': subtitles,
+            'title': meta['title'],
+            'description': meta['description'],
+            'thumbnail': meta['thumbnail']
```
Comment (dstftw) [inline youtube_dl/extractor/amara.py]: Read coding conventions on optional metadata.

File: youtube_dl/extractor/amara.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        video_url = meta.get('all_urls')[0]
+        subtitles = dict(map(lambda language: [
+            language['code'],
+            [
+                {
+                    'ext': 'vtt',
+                    'url': language['subtitles_uri'].replace('format=json', 'format=vtt')
+                }, {
+                    'ext': 'srt',
+                    'url': language['subtitles_uri'].replace('format=json', 'format=srt')
```
Comment (dstftw) [inline youtube_dl/extractor/amara.py]: DRY. Extract json format as well.

File: youtube_dl/extractor/amara.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .common import InfoExtractor
+
+
+class AmaraIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?amara\.org/(?:\w+/)?videos/(?P<id>\w+).*'
```
Comment (dstftw) [inline youtube_dl/extractor/amara.py]: `.*` is useless at the end.

**Comment by joostverdoorn** [comment]
@dstftw processed your comments!

**Comment by joostverdoorn** [comment]
@dstftw sorry to keep spamming, but it would be great if we could get this merged!

**Comment by joostverdoorn** [comment]
@dstftw great to have the project reinstated! Let's celebrate by merging this PR? 😇

**Pull request merged**